src/arviz_plots/plots/essplot.py

```diff
@@ -0,0 +1,585 @@
+        >>> )
+
+    The artists' visual features can also be customized through plot_kwargs, based on the
+    kwargs that the visual element function for the artist accepts- like all the other
+    batteries included plots. For example, for the 'ess' artist, the scatter_xy function is
+    used. So if we want to change the marker:
+
+    .. plot::
+        :context: close-figs
+
+        >>> pc = plot_ess(
+        >>>    centered,
+        >>>    coords={"school": ["Choate", "Deerfield", "Hotchkiss"]},
+        >>>    plot_kwargs={"ess": {"marker": "_"}},
+        >>> )
+
+    """
+    # initial defaults
+    if sample_dims is None:
+        sample_dims = rcParams["data.sample_dims"]
+    if isinstance(sample_dims, str):
+        sample_dims = [sample_dims]
+
+    ylabel = "{}"
+
+    # mutable inputs
+    if plot_kwargs is None:
+        plot_kwargs = {}
+    if pc_kwargs is None:
+        pc_kwargs = {}
+    else:
+        pc_kwargs = pc_kwargs.copy()
+
+    if stats_kwargs is None:
+        stats_kwargs = {}
+
+    # processing dt/group/coords/filtering
+    distribution = process_group_variables_coords(
+        dt, group=group, var_names=var_names, filter_vars=filter_vars, coords=coords
+    )
+
+    # ensuring plot_kwargs['rug'] is not False
+    rug_kwargs = copy(plot_kwargs.get("rug", {}))
+    if rug_kwargs is False:
+        raise ValueError("plot_kwargs['rug'] can't be False, use rug=False to remove the rug")
+
+    if backend is None:
+        backend = rcParams["plot.backend"]
+    plot_bknd = import_module(f".backend.{backend}", package="arviz_plots")
+
+    # set plot collection initialization defaults if it doesnt exist
+
+    # figsizing related plot collection initialization
+    if plot_collection is None:
+        figsize = pc_kwargs.get("plot_grid_kws", {}).get("figsize", None)
+        figsize_units = pc_kwargs.get("plot_grid_kws", {}).get("figsize_units", "inches")
+        pc_kwargs.setdefault("col_wrap", 5)
+        pc_kwargs.setdefault(
+            "cols",
+            ["__variable__"]
+            + [dim for dim in distribution.dims if dim not in {"model"}.union(sample_dims)],
+        )
+        n_plots, _ = process_facet_dims(distribution, pc_kwargs["cols"])
+        col_wrap = pc_kwargs["col_wrap"]
+        if n_plots <= col_wrap:
+            n_rows, n_cols = 1, n_plots
+        else:
+            div_mod = divmod(n_plots, col_wrap)
+            n_rows = div_mod[0] + (div_mod[1] != 0)
+            n_cols = col_wrap
+    else:
+        figsize, figsize_units = plot_bknd.get_figsize(plot_collection)
+        n_rows = leaf_dataset(plot_collection.viz, "row").max().to_array().max().item()
+        n_cols = leaf_dataset(plot_collection.viz, "col").max().to_array().max().item()
+
+    plot_bknd = import_module(f".backend.{backend}", package="arviz_plots")
+
+    figsize = plot_bknd.scale_fig_size(
+        figsize,
+        rows=n_rows,
+        cols=n_cols,
+        figsize_units=figsize_units,
+    )
```

Comment:
This is the initial version of figsize scaling, it should follow the same pattern as plot_forest. The only thing to take from plot_trace is the computation of number of rows and columns (instead of using the one from plot_forest which would make no sense here).

https://github.com/arviz-devs/arviz-plots/blob/main/src/arviz_plots/plots/forestplot.py#L258-L278, changing https://github.com/arviz-devs/arviz-plots/blob/main/src/arviz_plots/plots/forestplot.py#L271-L272 to the computation plot_trace does to get the number of columns and rows in the generated grid.